MediaBrowser.Common/Plugins/BasePlugin.cs

```diff
@@ -137,7 +134,26 @@ protected BasePlugin(IApplicationPaths applicationPaths, IXmlSerializer xmlSeria
                 var assemblyName = assembly.GetName();
                 var assemblyFilePath = assembly.Location;
 
-                var dataFolderPath = Path.Combine(ApplicationPaths.PluginsPath, Path.GetFileNameWithoutExtension(assemblyFilePath));
+                // Find out the plugin folder.
+                bool inPluginFolder = assemblyFilePath.StartsWith(ApplicationPaths.PluginsPath, StringComparison.OrdinalIgnoreCase);
+                string path, dataFolderPath;
+
+                var configurationFileName = Path.ChangeExtension(Path.GetFileName(assemblyFilePath), ".xml");
+                if (inPluginFolder)
+                {
+                    // Normal plugin.
+                    path = assemblyFilePath.Substring(ApplicationPaths.PluginsPath.Length).Split('\\', StringSplitOptions.RemoveEmptyEntries)[0];
+                    dataFolderPath = Path.Combine(
+                        Path.Combine(ApplicationPaths.PluginsPath, path),
+                        configurationFileName);
+                    ConfigurationFilePath = dataFolderPath;
```

Comment:
Does this change plugin config to be stored along with the plugin files?

Reply:
Yes - two things became apparent. 1) if there were multiple versions of the plugin, there could be multiple versions of the config. 2) dataFolderPath pointed to an invalid path if the foldername and the plugin name weren't identical.
(One of the users complained about this after he lost data due to an auto-upgrade)

A migration takes place if the config file exists in the configuration folder and not in the plugin folder, copy it across.

Reply:
The dataFolderPath issue is that the folder name has the version number appended to it now.
The original code doesn't see this so points to an invalid folder.

Reply:
Have amended PluginManager.UpdateSuccessors to copy across any xml files from the previous active installation if they don't exist. 

This along with the migration from the /configuration folder should enable changes in configurations to be kept between between versions as well as permitting each plugin to make changes, that won't be applied to other versions.

Reply:
I don't think this change is needed, and I think it will only cause issues / more confusion.
I just tested latest master and didn't have any issue saving / loading plugin configurations.

It's up to the plugin developer to not make breaking changes to the configuration.

Reply:
Fair enough - have rolled back change - but added a clause to append the version number onto the dataFolderPath if the folder doesn't exist (due to the versioning info now appended to the foldername)